### DISCUSSION TITLE
React Native: Apply Skip to all Queries using Redux State

### DISCUSSION THREAD

**Discussion by nicholascm** [reduxjs/redux-toolkit#3527] [Q&A]
Background: we render our RN app even while the user is in a "logged out" state because we work in a secure environment where we have to auth challenge every time they background the app. The main app is covered with an auth modal.

I'd like to be able to apply a `skip` parameter to all of our queries (they are generated from swagger.json), e.g. something like looking at the redux state to determine if the user is logged in and therefore if queries should send a network request or not, so as to avoid the failed request. 

One way that would work is just adding the `skip` token everywhere the hooks are used, but that is a bit cumbersome and repetitive. Any ideas on doing this at a higher level so it applies to all?

**Comment by markerikson** [comment]
I don't think we currently have a way to do blanket skips, or even skips at the plain-endpoint-only level - just via the hooks.

**Reply by nicholascm** [reply]
Thanks Mark.

**Reply by nicholascm** [reply]
I ended up going with something similar to this: https://github.com/reduxjs/redux-toolkit/issues/2916 to just hold my network requests until a state was satisfied.
